int32 velocity = 0
int32 position = 1
int32 position_velocity = 2